### PR TITLE
[superseded] Add support for ObjectId

### PR DESCRIPTION
**Pull request by nirinchev**
Fixes https://github.com/realm/realm-dart/issues/461.

### CODE REVIEW

File: common/pubspec.yaml [hunk 1]
```diff
@@ -9,6 +9,9 @@ homepage: https://www.realm.io
 repository: https://github.com/realm/realm-dart
 issue_tracker: https://github.com/realm/realm-dart/issues
 
+dependencies:
+  objectid: ^2.0.0
```
Comment (nielsenko) [inline common/pubspec.yaml:13]: Nice. I thought we had to roll over own.

File: lib/src/native/realm_core.dart [hunk 2]
```diff
@@ -854,7 +863,13 @@ extension on Pointer<realm_value_t> {
...
         throw Exception("Not implemented");
       case realm_value_type.RLM_TYPE_OBJECT_ID:
-        throw Exception("Not implemented");
+        final nativeBytes = ref.values.object_id.bytes;
+        final bytes = List<int>.filled(12, 0);
+        for (int i = 0; i < 12; i++) {
+          bytes[i] = nativeBytes[i];
+        }
+
+        return ObjectId.fromBytes(bytes);
```
Comment (nielsenko) [inline lib/src/native/realm_core.dart:872]:
```suggestion
        return ObjectId.fromBytes(cast<Uint8>().asTypedList(12));
```
The `ObjectId.fromBytes` ctor already copies the bytes. Given the memory layout of the `realm_value_t` we can just read the first 12 bytes.

**Review by nielsenko — APPROVED** [review]

File: generator/test/good_test_data/all_types.expected [hunk 3]
```diff
@@ -0,0 +1,232 @@
+// **************************************************************************
```
Comment (nirinchev) [inline generator/test/good_test_data/all_types.expected:1]: This is not strictly related to this PR, but I saw we don't validate the outputs of the `all_types.dart` test, which wasn't great as it would only validate that it doesn't error out, but not that it produces meaningful results. I wanted to verify the outputs for the new type, so I had to add the entire file.

**Comment by blagoev** [comment]
This was superseded by #468

**Comment by blagoev** [comment]
reopened since this has nullable objectid support

**Comment by nirinchev** [comment]
Covered by https://github.com/realm/realm-dart/pull/851
